mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -251,7 +252,7 @@ private boolean textContains(String value, MailboxMessage message) throws IOExce
     }
 
     private InputStream textHeaders(MailboxMessage message) throws MimeIOException, IOException {
-        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        ByteArrayOutputStream out = new ByteArrayOutputStream(Ints.checkedCast(message.getFullContentOctets()));
```

Comment:
Well, I intended to do this as a quick fix, and don't want to start changing every stream in James.

Here the idea is to avoid the call to the private method **grow** in **ByteArrayOutputStream**, not switch the implementation to something more efficient. Which is different.

Basically, ByteArrayOutput stream without size is bad, that's what I intended to quick-fix. That's different IMO from reworking the entire IO related code. If this is not desirable, let's forget.

However, I agree